enzyme/test/Enzyme/ReverseMode/blas/cblas_dscal_nomod.ll

```diff
@@ -0,0 +1,67 @@
+;RUN: %opt < %s %loadEnzyme -enzyme -mem2reg -instsimplify -simplifycfg -S | FileCheck %s
+
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+@__const.main.m = private unnamed_addr constant [3 x double] [double 1.000000e+00, double 2.000000e+00, double 3.000000e+00], align 16
+@__const.main.d_m = private unnamed_addr constant [3 x double] [double 1.000000e+00, double 1.000000e+00, double 1.000000e+00], align 16
+@.str = private unnamed_addr constant [10 x i8] c"%f %f %f\0A\00", align 1
+
+; Function Attrs: noinline nounwind optnone ssp uwtable
+define void @f(double* noalias %0) #0 {
+  %2 = alloca double*, align 8
+  store double* %0, double** %2, align 8
+  %3 = load double*, double** %2, align 8
+  call void @cblas_dscal(i32 3, double 2.000000e+00, double* %3, i32 1)
```

Comment:
Sure, this can be simplified (not sure why llvm didn't do that), and I agree that the double (scaling factor a) should be passed to f (so we can probably calculate the derivative of it), but what is the point of passing other integers (size and stride) to f?